lib/streamlit/elements/html.py

```diff
@@ -30,7 +31,7 @@ class HtmlMixin:
     @gather_metrics("html")
     def html(
         self,
-        body: str | Path,
+        body: str | Path | Any,
```

Comment:
Is this something we can leverage a [protocol](https://realpython.com/python-protocol/#static-duck-typing-with-protocols) instead of just allowing `Any`?

```python
from typing import Protocol

class ReprHtml(Protocol):
    def _repr_html_(self) -> str:
        ...
```

Reply:
Just realized you defined it in the type_util. Any reason we cannot have it instead of `Any`?

Reply:
Ah we also want to allow for `__str__`, then I guess it could be easy with `str | Path | object`?